### PR TITLE
project search app scope

### PR DESCRIPTION
**Pull request by D-Lite**
Implement projects search from the homepage

### CODE REVIEW

File: app/components/NavBar.js [hunk 1]
```diff
@@ -50,6 +50,27 @@ const NavBar = () => {
...
   const pathname = usePathname();
+  const [searchProjectsText, setSearchProjectsText] = useState('')
+
+  const router = useRouter();
+
+  const handleSearch = (e) => {
+    e.preventDefault();
+
+    // Redirect to the /project URL with the search term as a query parameter
+    router.push(`/projects?search=${encodeURIComponent(searchProjectsText)}`);
```
Comment (kaf-lamed-beyt) [inline app/components/NavBar.js:61]: @D-Lite I think you should use `router.replace()` here. So we don't end up populating the history stack on every search change 

You can be explicit about it too by including the `shallow` property and setting it to `true` like so:

```js
router.replace(`/projects?search=${encodeURIComponent(searchProjectsText)}`, undefined, {shallow: true})
```

More on that [here](https://nextjs.org/docs/pages/api-reference/functions/use-router#routerreplace)

Comment (D-Lite) [reply app/components/NavBar.js]: Yes, good performance improvement. 

Thanks @kaf-lamed-beyt

Comment (kaf-lamed-beyt) [reply app/components/NavBar.js]: Anytime, chief! 🍷

Comment (acekyd) [reply app/components/NavBar.js]: Thanks fellas @D-Lite @kaf-lamed-beyt

**Comment by acekyd** [comment]
Taking a few moments to investigate some issues:
- First time I hit enter on search through the navbar did not respond
- Some keywords return less or no results than they actually should. E.g laravel or feeder.

**Comment by D-Lite** [comment]
@acekyd, the search works well now.

The search key responds immediately, but might be a sec late before reflecting on the route,

**Comment by acekyd** [comment]
The search now works great on the projects page. But not on the home page and navbar components - the experience is quite subpar. 

I'll recommend we take out the search projects text field on both the nav bar and homepage - and change it to a button (with similar look) that says "Explore projects". This will redirect people to the projects page where they can then search. @D-Lite

**Comment by D-Lite** [comment]
That works perfectly @acekyd 

Let me know what you think about the new button! Thanks

**Comment by acekyd** [comment]
Nice one with the button - but also supposed to change the one on the home page as mentioned. At the moment, the homepage search breaks as well. @D-Lite

**Comment by D-Lite** [comment]
Updated @acekyd 

Thank you!

**Comment by acekyd** [comment]
Nice

**Pull request merged**